### DISCUSSION TITLE
Dependency injection in entry.server.tsx ?

### DISCUSSION THREAD

**Discussion by ajaishankar** [remix-run/remix#10332] [Q&A] [answered]
Referring to @sergiodxa's article on dependency injection.

https://sergiodxa.com/articles/dependency-injection-in-remix-loaders-and-actions

I want to use the built in remix server and not a custom express server.

Would it be ok to add dependencies to loadContext in `entry.server.tsx` like this?

```ts

const cache = new Cache();

export default function handleRequest(
  request: Request,
  responseStatusCode: number,
  responseHeaders: Headers,
  remixContext: EntryContext,
  loadContext: AppLoadContext
) {
  loadContext.cache = cache; // inject dependency
}
```

**Comment by sergiodxa** [comment, marked as answer]
That won't work, entry.server functions run after your loaders, it will be to late for that moment

You have to use a custom http server.

**Reply by ajaishankar** [reply]
Thanks!
